src/coreclr/CMakeLists.txt

```diff
@@ -1,16 +1,9 @@
-cmake_minimum_required(VERSION 3.6.2)
+cmake_minimum_required(VERSION 3.20)
 
 cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH is enabled by default.
```

Comment:
Do we still need to set these policies individually after specifying minimum required version 3.20?

`cmake_minimum_required(VERSION 3.20)` should set these policies to NEW according to the docs: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html

Reply:
We plan to update the `cmake_policy` statements, remove redundant ones, as well as other build improvements after doing the initial minimum version upgrade. We wanted to keep the PR that is most likely to break workflows as small as possible to limit the risk of the change (and risk of having to revert it).

Reply:
If you want to keep this PR as minimal as possible and keep cleanup for later, this block can stay as is. The `if (CMAKE_VERSION VERSION_GREATER ...)` conditions are harmless.